### DISCUSSION TITLE
Listbox: How to get value of Selected Item to pass to State?

### DISCUSSION THREAD

**Discussion by arkmech** [tailwindlabs/headlessui#2083] [Help]
**Background:** Passing React State Setter to onChange automatically sets the selected value to state.
```javascript
function MyListbox() {
  return (
    <Listbox value={selectedOption} onChange={setSelectedOption}>
   
    </Listbox>
  )
}
```

What I need is additional functionality when `<MyListbox/>` changes besides just setting state, therefore I take an `onChange` function.
```javascript
function MyListbox({onChange}) {
  return (
    <Listbox value={selectedOption} onChange={onChange}>
   
    </Listbox>
  )
}
```

`onChange` being taken in as a prop, how to get selectedOption value outside of `<MyListbox/>` component?
```javascript
onChange={() => {
  setSelectedOption();
  // other functionality
}}
```

**Comment by sboy99** [comment]
`Headless ui` itself handles the state by passing a value
```javascript
const [selectedOption,setSelectedOption]=useState(selectedOptions[0]);
```
The function `setSelectedOption` goes like this
```typescript
(value:Type)=>void
```
So you have to define the function as per the context
#### Solution
```typescript
`TypeScript`:

  onChange={(value:Type) => {
  setSelectedOption(value);
  // other functionality
}}

`JavaScript`: 

onChange={(value) => {
  setSelectedOption(value);
  // other functionality
}}
```
